pydantic/_internal/_generate_schema.py

```diff
@@ -542,6 +543,26 @@ def ser_ip(ip: Any) -> str:
             ),
         )
 
+    def _fraction_schema(self) -> CoreSchema:
+        """Support for [`fractions.Fraction`][fractions.Fraction]."""
+        from ._validators import fraction_validator
+
+        # TODO: note, this is a fairly common pattern, re lax / strict for attempted type coercion,
+        # can we use a helper function to reduce boilerplate?
+        return core_schema.lax_or_strict_schema(
+            lax_schema=core_schema.no_info_plain_validator_function(fraction_validator),
+            strict_schema=core_schema.json_or_python_schema(
+                json_schema=core_schema.no_info_plain_validator_function(fraction_validator),
+                python_schema=core_schema.is_instance_schema(Fraction),
+            ),
+            # use str serialization to guarantee round trip behavior
+            serialization=core_schema.to_string_ser_schema(when_used='always'),
+            # TODO: figure out if this is actually how to represent fractions in JSON schema
```

Comment:
```suggestion
```
I think this is okay, I believe we already do things that aren't a part of the spec

Reply:
Yeah, @Viicos fixed some of those cases recently

Reply:
https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-validation-00#rfc.section.7.3

> These attributes apply to string instances.
> 
> Date and time format names are derived from [RFC 3339, section 5.6](https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-validation-00#RFC3339). The duration format is from the ISO 8601 ABNF as given in Appendix A of RFC 3339.
> 
> Implementations supporting formats SHOULD implement support for the following attributes:
> 
> date-time:
> A string instance is valid against this attribute if it is a valid representation according to the "date-time" production.
> date:
> A string instance is valid against this attribute if it is a valid representation according to the "full-date" production.
> time:
> A string instance is valid against this attribute if it is a valid representation according to the "full-time" production.
> duration:
> A string instance is valid against this attribute if it is a valid representation according to the "duration" production.
>
> **Implementations MAY support additional attributes using the other production names defined anywhere in that RFC. If "full-date" or "full-time" are implemented, the corresponding short form ("date" or "time" respectively) MUST be implemented, and MUST behave identically. Implementations SHOULD NOT define extension attributes with any name matching an RFC 3339 production unless it validates according to the rules of that production. [CREF5]**

(Emphasis mine.)

If I'm interpreting correctly, the only ones that we would need to be careful about would be, from https://www.rfc-editor.org/rfc/rfc3339:

> [5.6](https://www.rfc-editor.org/rfc/rfc3339#section-5.6). Internet Date/Time Format
>
>    The following profile of ISO 8601 [[ISO8601](https://www.rfc-editor.org/rfc/rfc3339#ref-ISO8601)] dates SHOULD be used in
>    new protocols on the Internet.  This is specified using the syntax
>    description notation defined in [[ABNF](https://www.rfc-editor.org/rfc/rfc3339#ref-ABNF)].
>
>    date-fullyear   = 4DIGIT
>    date-month      = 2DIGIT  ; 01-12
>    date-mday       = 2DIGIT  ; 01-28, 01-29, 01-30, 01-31 based on
>                              ; month/year
>    time-hour       = 2DIGIT  ; 00-23
>    time-minute     = 2DIGIT  ; 00-59
>    time-second     = 2DIGIT  ; 00-58, 00-59, 00-60 based on leap second
>                              ; rules
>    time-secfrac    = "." 1*DIGIT
>    time-numoffset  = ("+" / "-") time-hour ":" time-minute
>    time-offset     = "Z" / time-numoffset
>
>    partial-time    = time-hour ":" time-minute ":" time-second
>                      [time-secfrac]
>    full-date       = date-fullyear "-" date-month "-" date-mday
>    full-time       = partial-time time-offset
>
>    date-time       = full-date "T" full-time

In particular, I think it's safe to just add whatever stuff we think makes sense.